test/test_inverse.jl

```diff
@@ -122,3 +124,15 @@ InverseFunctions.inverse(f::Bar) = Bar(inv(f.A))
         InverseFunctions.test_inverse(log ∘ foo, x)
     end
 end
+
+VERSION ≥ v"1.9" && @testset "dates" begin
```

Comment:
More directly, you could check whether extensions are supported since that's the feature the functionality depends on:
```suggestion
if isdefined(Base, :get_extension)
    @testset "dates" begin
```

Reply:
Extension support is also available in pre-releases of 1.9, so technically the check `VERSION >= v"1.9"` is not correct. Checking whether `Base.get_extension` is defined is the officially documented way for checking whether a Julia version supports extensions or not. If you are concerned about it being removed in future Julia versions, you could combine it with a version check (see, e.g., https://github.com/JuliaStats/Distances.jl/pull/246#discussion_r1335666792).